tests/unit_test/gmail_data/get_gmail_data_test.py

```diff
@@ -0,0 +1,219 @@
+
+        assert actual_messages == ['messages1', 'messages_in_next_page1']
+
+
+class TestGetDataframeForThreadResponse:
+
+    def test_should_pass_user_id_and_import_timestamp_in_method(self):
+        thread_response = {"id": "id1", "historyId": "historyId1", "messages": "messages1"}
+        df = get_dataframe_for_thread_response(thread_response, USER_ID1, IMPORTED_TIMESTAMP)
+        assert df['user_id'].values == [USER_ID1]
+        assert df['imported_timestamp'].values == [IMPORTED_TIMESTAMP]
+
+    def test_should_return_elements_in_thread_response(self):
+        thread_response = {"id": "id1", "historyId": "historyId1", "messages": "messages1"}
+        df = get_dataframe_for_thread_response(thread_response, USER_ID1, IMPORTED_TIMESTAMP)
+        assert df['threadId'].values == ['id1']
+        assert df['historyId'].values == ['historyId1']
+        assert df['messages'].values == ['messages1']
+
+
+class TestGetOneThread:
+
+    def test_should_pass_user_id_and_thread_id_to_get_method(
+            self,
+            gmail_service_mock,
+            gmail_thread_get_mock):
+
+        get_one_thread(gmail_service_mock, USER_ID1, THREAD_ID1)
+        gmail_thread_get_mock.assert_called_with(userId=USER_ID1, id=THREAD_ID1)
+
+
+class TestIterGmailHistory:
+
+    def test_should_pass_user_id_and_start_history_id_to_history_list_method(
+            self,
+            gmail_service_mock,
+            gmail_history_list_mock):
+
+        list(iter_gmail_history(gmail_service_mock, USER_ID1, START_ID1))
+        gmail_history_list_mock.assert_called_with(userId=USER_ID1, startHistoryId=START_ID1)
+
+    def test_should_yield_first_page_history_only_without_next_page_token(
+            self,
+            gmail_service_mock,
+            gmail_history_list_mock):
+
+        response = {'history': ['history1']}
+
+        execute_mock = gmail_history_list_mock.return_value.execute
+        execute_mock.return_value = response
+
+        actual_messages = list(iter_gmail_history(gmail_service_mock, USER_ID1, START_ID1))
+        assert actual_messages == ['history1']
+
+    def test_should_next_page_history_if_exists_next_page_token(
+            self,
+            gmail_service_mock,
+            gmail_history_list_mock):
+
+        response = {'history': ['history1'], 'nextPageToken': 'nextPageToken1'}
+        response_in_next_page = {'history': ['history_in_next_page1']}
+
+        execute_mock = gmail_history_list_mock.return_value.execute
+        execute_mock.side_effect = [response, response_in_next_page]
+
+        actual_messages = list(iter_gmail_history(gmail_service_mock, USER_ID1, START_ID1))
+
+        assert actual_messages == ['history1', 'history_in_next_page1']
+
+
+class TestGetDataframeForHistoryResponse:
+
+    def test_get_dataframe_for_history_one_response(self):
+        history_response = [{'id': 'id1', 'messages': [{'id': 'id2', 'threadId': 'threadId1'}]}]
```

Comment:
related to https://github.com/elifesciences/data-hub-core-airflow-dags/pull/452/files#r617483057 the name `history_list` or similar might be slightly better?